integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml

```diff
@@ -107,8 +107,9 @@ spec:
           envFrom:
           - configMapRef:
               name: {{ $fullName }}-config
-{{ include "alluxio.master.readinessProbe" . | indent 10 }}
-{{ include "alluxio.master.livenessProbe" . | indent 10 }}
+{{- $probePort := $isHaEmbedded | ternary "embedded" "rpc" }}
+{{ include "alluxio.master.readinessProbe" (dict "port" $probePort) | indent 10 }}
+{{ include "alluxio.master.livenessProbe" (dict "port" $probePort) | indent 10 }}
```

Comment:
We cannot use RPC port as the indicator for HA (embedded journal), because only the leader's port will accept connections. All followers will refuse connection on both RPC and web port, so the probes won't work.

In `alluxio-monitor.sh master`, it uses `MasterHealthCheckClient` https://github.com/Alluxio/alluxio/blob/164cf8636a145391591bd9dab9cb102a3f6921a3/shell/src/main/java/alluxio/master/MasterHealthCheckClient.java#L205.
The logic there is to SSH and look for the master/job-master process on each node. If the processes are all there, check if the primary's RPC port is ready. This cannot be done easily in the probe.